src/TumblThree/TumblThree.Applications/Downloader/FileDownloader.cs

```diff
@@ -131,6 +131,8 @@ public async Task<bool> DownloadFileWithResumeAsync(string url, string destinati
                         }
                         else
                         {
+                            fileStream?.Dispose();
+                            File.Delete(destinationPath);
```

Comment:
I would not delete the file in all cases. What about a video download that timeouts short before the end? It prevents resume functionality.

Reply:
Strangely enough, as of now, I don't see resume to be working, probably due to skipping any existing files in filesystem or DB (index).